src/core/cgroup-bpf-ctx.c

```diff
@@ -0,0 +1,58 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#include "cgroup-bpf-ctx.h"
+#include "cgroup.h"
+#include "memory-util.h"
+#include "strv.h"
+#include "string-util.h"
+
+int cgroup_add_pinned_bpf_program(CGroupContext *c, enum bpf_attach_type attach_type, const char *bpffs_path) {
+        _cleanup_free_ CGroupPinnedBPFProgram *p = NULL;
+        _cleanup_free_ char *dup = NULL;
+
+        assert(c);
+        assert(bpffs_path);
+
+        if (cgroup_contains_pinned_bpf_program(c, attach_type, bpffs_path))
+                return log_warning_errno(SYNTHETIC_ERRNO(EEXIST),
+                                "BPF program %s is already specified in cgroup context", bpffs_path);
+
+        dup = strdup(bpffs_path);
+        if (!dup)
+                return log_oom();
+
+        p = new(CGroupPinnedBPFProgram, 1);
+        if (!p)
+                return log_oom();
+
+        *p = (CGroupPinnedBPFProgram) {
+                .attach_type = attach_type,
+                .bpffs_path = TAKE_PTR(dup),
+        };
+
+        LIST_PREPEND(prog, c->pinned_bpf_progs, TAKE_PTR(p));
+
+        return 0;
+}
+
+bool cgroup_contains_pinned_bpf_program(
+                CGroupContext *c, enum bpf_attach_type attach_type, const char *bpffs_path) {
+        CGroupPinnedBPFProgram *p, *n;
+
+        assert(c);
+        assert(bpffs_path);
+
+        LIST_FOREACH_SAFE(prog, p, n, c->pinned_bpf_progs)
```

Comment:
a regular LIST_FOREACH() should suffice.

LIST_FOREACH_SAFE() is only useful if you intend to modify the list while you iterate through it

Reply:
A comment from an earlier review: https://github.com/systemd/systemd/pull/13496#discussion_r546154729

Should applications of `LIST_FOREACH_SAFE` vs `LIST_FOREACH` be mentioned in style guide?
Switched back to `LIST_FOREACH`.

Reply:
i figure the list.h header file should carry a comment about that next to the _SAFE macro definition...